.github/workflows/promote_release.yaml

```diff
@@ -0,0 +1,191 @@
+#
+# It will release the staged maven artefacts, and merge the release PR.
+#
+# It requires the following:
+# inputs:
+# release-pr-issue-number - the issue number of release PR
+# command - promote-release or drop-release
+#
+# variables:
+# KROXYLICIOUS_SONATYPE_TOKEN_USERNAME - Sonatype Access User Token Username
+#
+# secrets:
+# KROXYLICIOUS_SONATYPE_TOKEN_PASSWORD - Sonatype Access User Token Password
+# KROXYLICIOUS_RELEASE_TOKEN             - GitHub PAT wih content/createPullRequest permission for Kroxylicious repo.
+#
+
+on:
+  workflow_call:
+    inputs:
+      release-pr-issue-number:
+        description: 'PR number that is performing the release'
+        type: number
+        required: true
+      command:
+        description: 'Dispatched command'
+        required: true
+        type: string
+    secrets:
+      KROXYLICIOUS_SONATYPE_TOKEN_PASSWORD:
+        required: true
+      KROXYLICIOUS_RELEASE_TOKEN:
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'Check out repository'
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+          fetch-tags: true
+          fetch-depth: 0
+
+      - name: 'Check team membership'
+        if: ${{ github.repository == 'kroxylicious/kroxylicious' }}
+        uses: tspascoal/get-user-teams-membership@v2
+        id: team-membership
+        with:
+          username: ${{ github.actor }}
+          team: release-engineers
+          GITHUB_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+
+      - name: 'Stop workflow if user is not a release-engineer'
+        if: ${{ github.repository == 'kroxylicious/kroxylicious' && steps.team-membership.outputs.isTeamMember == 'false' }}
+        run: |
+          gh issue comment  ${{ github.event.issue.number }} --body "${{ github.actor }} is not a member of https://github.com/orgs/kroxylicious/teams/release-engineers)"
+          exit -1
+
+      - name: 'Get the branch name associated with the PR'
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+            echo "PR_BRANCH=$(gh pr view ${{ inputs.release-pr-issue-number}} --json headRefName | jq -r .headRefName)" >> $GITHUB_ENV
+
+      - name: 'Branch name matches naming conventions'
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+          if [[ ! "${PR_BRANCH}" =~ ^release-work- ]]; then
+            gh issue comment  ${{ github.event.issue.number }} --body "This PR's branch name (${PR_BRANCH}) is not in the expected form."
+            exit -1
+          fi
+
+      - name: 'Get release version from the branch of PR'
+        run: |
+          # This knows the commit that precedes PR_BRANCH will carry the tag.
+          TAG=$(git name-rev --no-undefined --tags --name-only  origin/${PR_BRANCH}^)
+          echo "RELEASE_VERSION=$(echo ${TAG} | sed -e 's/^v//')" >> $GITHUB_ENV
```

Comment:
I think the sed pattern deserves a comment (mostly because I don't remember what `^v` does off the top of my head and struggle to visualise what its ensuring. 

Though I do wonder if it should be sanitising TAG earlier to make the echo simpler... (that could well be point less shuffling)

Reply:
Version tags have a `v` prefix, so it needs that part removed.